qringbuf.go

```diff
@@ -1,3 +1,191 @@
+// Package qringbuf provides a concurrency-friendly zero-copy abstraction of
+// io.ReadAtLeast(…) over a pre-allocated ring-buffer, populated asynchronously
+// by a standalone goroutine. It is primarily designed for processing a series
+// of arbitrary streams each comprised of variable-length records. See the
+// Examples section and the additional example at StartFill(…) further down.
+//
+// Specifically an object of this package makes the following guarantees:
+//  • Memory is allocated only at construction time, never during streaming
+//  • StartFill(…) spawns off a single goroutine (collector) which terminates when:
+//    ◦ It reaches readLimit, if one was supplied
+//    ◦ It receives any error from the wrapped reader (including io.EOF)
+//    ◦ It had a chance to evaluate a signal received from StopFill()
+//      (most io.Read() operations are not cancelable, and may block forever)
+//  • Every call to NextRegion(…) blocks until it can return:
+//    ◦ ( *Region object representing a contiguous slice of at least MinRegion bytes, nil )
+//    ◦ ( *Region object representing all remaining data, any underlying error including io.EOF )
+//    ◦ ( nil when there is no data remaining in buffer, any underlying error including io.EOF )
+//  • *Region.Bytes() is always a slice of the underlying buffer, no data copying takes place
+//  • Data backing a *Region is guaranteed to remain available / not be overwritten provided:
+//    ◦ NextRegion(…) has not been called again allowing further writes into the buffer
+//    ◦ *Region.Reserve() was invoked, which blocks writes until a *Region.Release()
+//
+// Examples
+//
+// In code the basic usage looks roughly like this:
+//
+//  qrb, initErr := qringbuf.NewFromReader( *someReader, qringbuf.Config{ … } )
```

Comment:
I did and added the spaces afterwards, because of it rendering weird in a browser. The fact that you noticed this, means it is jarring for others too, so reverting all the post-formatting 😿 

https://github.com/ipfs/go-qringbuf/pull/6/commits/ee59439441fd